src/pages/ValidateLogin2FANewWorkspacePage.js

```diff
@@ -62,7 +66,12 @@ class ValidateLogin2FANewWorkspacePage extends Component {
             // by calling dismissModal(), the /v/... route is removed from history so the user will get taken to `/`
             // if they cancel out of the new workspace modal.
             Navigation.dismissModal();
-            Navigation.navigate(ROUTES.WORKSPACE_NEW);
+
+            if (this.props.betas) {
```

Comment:
If we are saving this behavior for the beta, why wouldn't we use

```suggestion
            if (Permissions.canUseFreePlan(this.props.betas)) {
```
And similarily for other parts referencing `this.props.betas`

Otherwise, aren't we saying this can be used by anyone? When betas load they'll be `[]` for those who aren't in any beta so the else part here wouldn't run.

Reply:
Well ROUTES.WORKSPACE_NEW already does the check for `Permissions.canUseFreePlan` so to keep the code DRY i think its best that the check for `canUseFreePlan` happens only in [NewWorkSpace.js](https://github.com/Expensify/App/blob/b8bf58ad6e7281b3b7342a1eb55ab747b9252513/src/pages/workspace/NewWorkspacePage.js#L47)

> Otherwise, aren't we saying this can be used by anyone? When betas load they'll be [] for those who aren't in any beta so the else part below here wouldn't run.

I didn't follow your question. Maybe this helps. The if condition determines if newWorkSpace component should render now or later. This take care of race conditions. If we have loaded betas, even if its empty, then we call  `Navigation.navigate(ROUTES.WORKSPACE_NEW);` to render NewWorkSpace since there is no reason to wait. If betas is null, that means we're still waiting and hence we call `setRedirectToWorkspaceNewAfterSignIn(true)` so that once the betas are loaded (even if its empty) we call Navigation.navigate(ROUTES.WORKSPACE_NEW); at a later time over [here](https://github.com/Expensify/App/pull/4205/files#diff-2a28868769d50b974461521d1d323f02f49c907ea488c6e79a7b058a8041017eR141). Does that make sense?

Reply:
Ahh yes that totally makes sense. Thanks for exposing that!